test/test_discovery_and_monitoring.py

```diff
@@ -400,9 +410,11 @@ def wait_for_thread(self, name):
         """Run the 'waitForThread' operation."""
         thread = self.targets[name]
         thread.stop()
-        thread.join()
+        thread.join(60)
```

Comment:
Was this timing out?

Reply:
Yes, see https://jira.mongodb.org/browse/PYTHON-2332.